becquerel/io/h5.py

```diff
@@ -1,7 +1,8 @@
 """Simple tools to perform HDF5 I/O."""
 
+from __future__ import annotations
```

Comment:
I didn't know this would enable the `|` syntax, really cool

Reply:
This should be due to [`rule UP007: non-pep604-annotation`](https://docs.astral.sh/ruff/rules/non-pep604-annotation/).